test/distributed/org/apache/cassandra/fuzz/topology/TopologyMixupTestBase.java

```diff
@@ -0,0 +1,697 @@
+                    possible.put(rs -> multistep(hostReplace(rs, state), waitForCMSToQuiesce()), 1);
+                    break;
+                default:
+                    throw new UnsupportedOperationException(task.name());
+            }
+        }
+        return Gens.oneOf(possible);
+    }
+
+    private static IntHashSet asSet(int[] array)
+    {
+        IntHashSet set = new IntHashSet(array.length);
+        for (int i : array)
+            set.add(i);
+        return set;
+    }
+
+    public interface SchemaSpec
+    {
+        String name();
+
+        String keyspaceName();
+    }
+
+    protected static class State<S extends SchemaSpec> implements AutoCloseable
+    {
+        final TopologyHistory topologyHistory;
+        final Cluster cluster;
+        final S schemaSpec;
+        final List<Runnable> preActions = new CopyOnWriteArrayList<>();
+        final AtomicLong currentEpoch = new AtomicLong();
+        final BiFunction<RandomSource, State<S>, Command<State<S>, Void, ?>> statementGen;
+        final Gen<RemoveType> removeTypeGen;
+        private final Map<String, Object> yamlConfigOverrides;
+        int[] cmsGroup = new int[0];
+
+        public State(RandomSource rs, BiFunction<RandomSource, Cluster, S> schemaSpecGen, Function<S, BiFunction<RandomSource, State<S>, Command<State<S>, Void, ?>>> cqlOperationsGen)
+        {
+            this.topologyHistory = new TopologyHistory(rs.fork(), 2, 4);
+            try
+            {
+
+                this.yamlConfigOverrides = CONF_GEN.next(rs);
+                cluster = Cluster.build(topologyHistory.minNodes)
+                                 .withTokenSupplier(topologyHistory)
+                                 .withConfig(c -> {
+                                     c.with(Feature.values())
+                                      .set("write_request_timeout", "10s");
+                                     //TODO (mainatiance): where to put this?  Anything touching ConfigGenBuilder with jvm-dtest needs this...
+                                     ((InstanceConfig) c).remove("commitlog_sync_period_in_ms");
+                                     for (Map.Entry<String, Object> e : yamlConfigOverrides.entrySet())
+                                         c.set(e.getKey(), e.getValue());
+                                     onConfigure(c);
+                                 })
+                                 //TODO (maintance): should TopologyHistory also be a INodeProvisionStrategy.Factory so address information is stored in the Node?
+                                 //TODO (maintance): AbstractCluster's Map<Integer, NetworkTopology.DcAndRack> nodeIdTopology makes playing with dc/rack annoying, if this becomes an interface then TopologyHistory could own
+                                 .withNodeProvisionStrategy((subnet, portMap) -> new INodeProvisionStrategy.AbstractNodeProvisionStrategy(portMap)
+                                 {
+                                     {
```

Comment:
nit: probably extra brackets are not necessary

Reply:
in this case not true.  This is the constructor